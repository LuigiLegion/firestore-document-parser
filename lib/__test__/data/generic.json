{
  "name": "projects/{project}/databases/(default)/documents/{collection}/{id}",
  "fields": {
    "double": {
      "doubleValue": 3.14
    },
    "array": {
      "arrayValue": {
        "values": [
          {
            "stringValue": "arrayString"
          }
        ]
      }
    },
    "null": {
      "nullValue": null
    },
    "reference": {
      "referenceValue": "projects/{project}/databases/(default)/documents/{collection}/{id}"
    },
    "map": {
      "mapValue": {
        "fields": {
          "mapString": {
            "stringValue": "mapString"
          }
        }
      }
    },
    "string": {
      "stringValue": "name"
    },
    "geopoint": {
      "geoPointValue": {
        "latitude": 40.7531823,
        "longitude": -73.9844421
      }
    },
    "integer": {
      "integerValue": "42"
    },
    "timestamp": {
      "timestampValue": "2021-01-31T17:34:56.123Z"
    },
    "boolean": {
      "booleanValue": true
    }
  },
  "createTime": "2021-01-31T20:30:43.030979Z",
  "updateTime": "2021-01-31T20:30:43.030979Z"
}
