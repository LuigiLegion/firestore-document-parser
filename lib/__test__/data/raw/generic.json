{
  "name": "projects/{project}/databases/(default)/documents/{collection}/{id}",
  "fields": {
    "double": {
      "doubleValue": 3.14
    },
    "array": {
      "arrayValue": {
        "values": [
          {
            "stringValue": "Fizz"
          },
          {
            "stringValue": "Buzz"
          },
          {
            "stringValue": "FizzBuzz"
          }
        ]
      }
    },
    "null": {
      "nullValue": null
    },
    "reference": {
      "referenceValue": "projects/{project}/databases/(default)/documents/{collection}/{id}"
    },
    "map": {
      "mapValue": {
        "fields": {
          "pangram": {
            "stringValue": "The quick brown fox jumps over the lazy dog"
          }
        }
      }
    },
    "string": {
      "stringValue": "Hello, World!"
    },
    "geopoint": {
      "geoPointValue": {
        "latitude": 40.7531823,
        "longitude": -73.9844421
      }
    },
    "integer": {
      "integerValue": "42"
    },
    "timestamp": {
      "timestampValue": "2021-01-31T12:34:56.789Z"
    },
    "boolean": {
      "booleanValue": true
    }
  },
  "createTime": "2021-01-31T20:30:43.030979Z",
  "updateTime": "2021-01-31T20:30:43.030979Z"
}
