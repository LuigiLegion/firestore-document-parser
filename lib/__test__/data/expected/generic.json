{
  "double": 3.14,
  "array": ["Fizz", "Buzz", "FizzBuzz"],
  "null": null,
  "reference": "projects/{project}/databases/(default)/documents/{collection}/{id}",
  "map": { "hello": "world" },
  "string": "The quick brown fox jumps over the lazy dog. If the dog reacted, was it really lazy?",
  "geopoint": {
    "latitude": 40.7531823,
    "longitude": -73.9844421
  },
  "integer": 42,
  "timestamp": "2021-01-31T17:34:56.123Z",
  "boolean": true
}
